### PR TITLE
[AIRFLOW-5843] Add conf option to Add DAG Run view

### PR DESCRIPTION
**Pull request by JCoder01**
---
Issue link: [AIRFLOW-5843](https://issues.apache.org/jira/browse/AIRFLOW-5843)

Adds input field to allow passing of conf when adding a dag run in the UI.

image

image

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by ashb** [comment]
This has no tests, and there should also be a check to ensure that the `conf` field is valid JSON.

**Comment by JCoder01** [comment]
@ashb is there a preferred way to validate the JSON or should I just add a try except with something like "Conf JSON isn't valid"?

**Comment by ashb** [comment]
@JCoder01 Yeah a try/except is probably good enough.

**Comment by JCoder01** [comment]
@ashb Ok, can you have a look?

File: airflow/www/validators.py [hunk 1]
```diff
@@ -56,3 +58,23 @@ def __call__(self, form, field):
...
+        self.message = message
+
+    def __call__(self, form, field):
+        if field.data:
+            try:
+                json.loads(field.data)
+            except Exception:
+                message = self.message or "{} is not valid JSON"
+                raise ValidationError(
+                    message=field.gettext(message.format(field.data))
```
Comment (ashb) [inline airflow/www/validators.py:79]: This message might be very long -- I don't know how exactly it's rendered, but I think including the content of the exception (from line 76) might be more useful -- for example that is `'Expecting value: line 3 column 1 (char 3)'`

**Review by ashb — COMMENTED** [review]
Other than the exception message this looks good.

**Review by ashb — APPROVED** [review]

**Review by kaxil — APPROVED** [review]

**Comment by kaxil** [comment]
Good work @JCoder01

**Pull request merged**